src/zenml/cli/stack_components.py

```diff
@@ -235,22 +238,133 @@ def register_stack_component_command(
                 cli_utils.error(str(e))
                 return
 
-            from zenml.stack.stack_component_class_registry import (
-                StackComponentClassRegistry,
-            )
+        from zenml.stack.registry import flavor_registry
 
-            component_class = StackComponentClassRegistry.get_class(
-                component_type=component_type, component_flavor=flavor
+        zenml_flavors = flavor_registry.get_flavors_by_type(
+            component_type=component_type
+        )
+
+        try:
+            # Try to find if there are any custom flavor implementations
+            flavor_wrapper = Repository().zen_store.get_flavor_by_name_and_type(
+                flavor_name=flavor,
+                component_type=component_type,
             )
-            component = component_class(name=name, **parsed_args)
+
+            # If there is one, check whether the same flavor exists as a default
+            # flavor to give out a warning
+            if flavor in zenml_flavors:
+                cli_utils.warning(
+                    f"The flavor that you are currently using "
+                    f"'{flavor_wrapper.name}' is overwriting the a flavor with "
+                    f"the same which is provided by ZenML. This could lead "
+                    f"to an unexpected behavior and should be avoided if "
+                    f"possible."
+                )
+
+        except KeyError:
+            if flavor in zenml_flavors:
+                flavor_wrapper = zenml_flavors[flavor]
+            else:
+                raise KeyError(
+                    f"There is no flavor '{flavor}' for the type "
+                    f"{component_type}"
+                )
+
+        try:
+            component = flavor_wrapper.to_flavor()(name=name, **parsed_args)
             Repository().register_stack_component(component)
             cli_utils.declare(
                 f"Successfully registered {display_name} `{name}`."
             )
 
+        except (ModuleNotFoundError, ImportError, NotImplementedError):
+            if flavor_wrapper.integration:
+                cli_utils.error(
+                    f"The {component_type} flavor '{flavor_wrapper.name}' is "
+                    f"a part of ZenML's '{flavor_wrapper.integration}' "
+                    f"integration, which is currently not installed on your "
+                    f"system. You can install it by executing: 'zenml "
+                    f"integration install {flavor_wrapper.integration}'."
+                )
```

Comment:
This error message is the sparkling jewel on top of this whole treasure of a PR ! I've been dreaming of having this for a long time now. Truly happy to see it come true.